docs/pages/web/tableofcontents.js

```diff
@@ -0,0 +1,186 @@
+      </MainSection>
+
+      <MainSection name="Subcomponents">
+        <MainSection.Subsection
+          title={generatedDocGen.TableOfContentsItem?.displayName}
+          description={generatedDocGen.TableOfContentsItem?.description}
+        >
+          <GeneratedPropTable
+            name={generatedDocGen.TableOfContentsItem?.displayName}
+            id={generatedDocGen.TableOfContentsItem?.displayName}
+            generatedDocGen={generatedDocGen.TableOfContentsItem}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <AccessibilitySection
+        name={generatedDocGen?.TableOfContents.displayName}
+        description="Be sure to include an `accessibilityLabel` for the screen reader for each item in the TableOfContents"
+      />
+
+      <MainSection
+        name="Localization"
+        description="Be sure to localize text and `accessibilityLabel`. Note that localization can lengthen text by 20 to 30 percent."
+      />
+
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Nested directory"
+          description="TableOfContents supports 5 levels of nesting. The first level maps to a section’s heading, which is usually an H3. The second level maps to a section’s subheading, which is usually an H4."
+        >
+          <MainSection.Card
+            cardSize="lg"
+            sandpackExample={
+              <SandpackExample name="Sticky header & footer example" code={nestedItemsExample} />
+            }
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="With heading"
+          description="A Heading can be added to TableofContents to be more clear about what is being navigated through."
+        >
+          <MainSection.Card
+            cardSize="lg"
+            sandpackExample={
+              <SandpackExample name="Sticky header & footer example" code={withHeaderExample} />
+            }
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <MainSection
+        name="Writing"
+        description="Items for a TableOfContents will be inherited from the headings on the page. For guidelines on writing headlines and titles, [see our Content Standards](https://gestalt.pinterest.systems/foundations/content_standards/voice)"
+      />
+
+      <QualityChecklist component={generatedDocGen?.TableOfContents.displayName} />
+
+      <MainSection name="Related">
+        <MainSection.Subsection
+          description={`
+**[SideNavigation](/web/SideNavigation)**
+SideNavigation is start-aligned and arranged vertically. It is used to navigate between page urls or sections when you have too many menu items to fit in horizontal [Tabs](/web/tabs).
+
+**[Tabs](/web/tabs)**
+Tabs may be used navigate between multiple URLs. Tabs are intended as page-level navigation.
+    `}
+        />
+      </MainSection>
+    </Page>
+  );
+}
+
+export async function getServerSideProps(): Promise<{|
```

Comment:
Override the Flow types for documentation purposes

<img width="875" alt="Screenshot by Dropbox Capture" src="https://github.com/pinterest/gestalt/assets/10593890/03e09e62-cb8e-4613-b6aa-0e13755d75c7">

Reply:
I don't think it should be aligned as the design on Figma is like that. But I will check with Hector

Reply:
Go to the doc page and follow this pattern to override the Flow types. We autogenerate Flow types from Flow but we can override them doing this: 
![image](https://github.com/pinterest/gestalt/assets/10593890/4b386617-e861-4bc5-a1ea-65a7a4e8fbe9)